From 58a1b34287bed2bf0a59073652967ca4018de875 Mon Sep 17 00:00:00 2001
From: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
Date: Mon, 17 Aug 2020 17:47:26 +0530
Subject: [PATCH 39/39] Add SGTL5000 audio codec support

This patch will enable SGTL5000 I2S audio codec support for
LEC-BASE 2.0 SMARC carrier

Signed-off-by: Antony Abee Prakash XV <antonyabee.prakashxv@adlinktech.com>
---
 arch/arm/boot/dts/lec-imx6.dtsi | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boot/dts/lec-imx6.dtsi b/arch/arm/boot/dts/lec-imx6.dtsi
index 2188564a8318..8ed85bc12503 100644
--- a/arch/arm/boot/dts/lec-imx6.dtsi
+++ b/arch/arm/boot/dts/lec-imx6.dtsi
@@ -6,7 +6,7 @@
 
 /* On-module multiplexer: 0 enables SMARC SPI0, !=0 enables I2S0 instead:
  * 1=TI TLV320AIC23 on LEC-Base R1 carrier, 2=Freescale SGTL5000 on LEC-Base R2 */
-#define CONFIG_I2S_AUDIO       1
+#define CONFIG_I2S_AUDIO       2
 
 /* The LEC-Base carrier doesn't include cameras, but allows connecting some. */
 #define CONFIG_SER_CAMERA      3c /* I2C addr of serial camera, hex without "0x" */
@@ -343,7 +343,6 @@
 &audmux {	/* SMARC I2S0 (instead of SPI0) */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_audmux>;
-	pinctrl-assert-gpios = <&gpio1 3 GPIO_ACTIVE_HIGH>; /* AEN# */
 	status = "okay";
 };
 #endif
@@ -374,7 +373,6 @@
 		   <&gpio3 24 GPIO_ACTIVE_LOW>; /* SMARC CS1 */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_ecspi1>;
-	pinctrl-assert-gpios = <&gpio1 3 GPIO_ACTIVE_LOW>; /* AEN# */
 	status = "okay";
 
 	tsc2046@0 {             /* Touch screen controller on LEC-Base R1 carrier */
@@ -687,9 +685,9 @@
 #endif
 
 #if CONFIG_I2S_AUDIO == 2
-	codec: sgtl5000@0a {            /* Audio codec on LEC-Base R2 carrier */
+	codec: sgtl5000@2a {            /* Audio codec on LEC-Base R2 carrier */
 		compatible = "fsl,sgtl5000";
-		reg = <0x0a>;
+		reg = <0x2a>;
 		clocks = <&clks IMX6QDL_CLK_CKO>; /* SMARC AUDIO_MCK, actually unused on carrier */
 		VDDA-supply = <&reg_base_3v3>;
 		VDDIO-supply = <&reg_base_1v8>;
@@ -771,7 +769,6 @@
 				/* Native pad functions */
 				MX6QDL_PAD_GPIO_0__CCM_CLKO1    0x130b0
 				/* Pads used for GPIOs */
-				MX6QDL_PAD_GPIO_3__GPIO1_IO03   0x80000000 /* AEN# (mux: 0=SPI0, 1=I2S0) */
 				MX6QDL_PAD_ENET_TXD1__GPIO1_IO29 0x80000000 /* GPIO_INT_B */
 				MX6QDL_PAD_ENET_TXD0__GPIO1_IO30 0x80000000 /* SMARC LCD_VDD_EN */
 				MX6QDL_PAD_ENET_RXD0__GPIO1_IO27 0x1a8b0 /* SMARC USB0_EN_OC# (open-drain) */
-- 
2.17.1

